### PR TITLE
Enable efi, clk and peterz trees

### PR DESCRIPTION
**Pull request by musamaanjum**
(see individual commits' description for information)

### CODE REVIEW

File: config/pipeline.yaml [hunk 1]
```diff
@@ -1913,6 +1934,14 @@ build_configs:
     branch: 'for-linus'
     variants: *build-variants
 
+  clk:
+    tree: clk
+    branch: 'clk-next'
+
```
Comment (a-wai) [inline config/pipeline.yaml:1940]:
```suggestion
    variants: *build-variants

```

For consistency (to be added to other build config entries as well).

File: config/pipeline.yaml [hunk 2]
```diff
@@ -117,6 +117,18 @@ _anchors:
       - 'amlogic'
       - 'ardb'
       - 'arnd'
+      - 'clk'
+      - 'peterz'
+
+  build-only-trees: &build-only-trees-rules-minimal
```
Comment (a-wai) [inline config/pipeline.yaml:123]:
```suggestion
  build-only-trees-minimal: &build-only-trees-minimal-rules
```

The entry has the same name as the previous one (`build-only-trees`) which should be avoided. Moreover, then anchor name should end with the "kind", in this case `-rules`.

I'm not a fan of having that many duplicated entries. If it's really necessary to have 2 sets of build-only trees, I'd rather have those as separate entries; although it feels better to have a single set and simply exclude `efi` adding a simple rule to scheduler entries where it shouldn't be used.

**Review by a-wai — CHANGES_REQUESTED** [review]

File: config/pipeline.yaml (hunk 1 above)
Comment (musamaanjum) [reply config/pipeline.yaml:1940]: @a-wai I've not been specifying variants for newly added trees from the time when it was indicated that these variants are legacy way of specifying them and we should avoid them. Why should variant be specified for these trees?

File: config/pipeline.yaml (hunk 2 above)
Comment (musamaanjum) [reply config/pipeline.yaml]: I'd not thought of it like this. I've updated as you have asked.

**Review by a-wai — APPROVED** [review]

File: config/pipeline.yaml [hunk 3]
```diff
@@ -1664,6 +1676,9 @@ scheduler:
 
   - job: kbuild-gcc-12-riscv-build-only
     <<: *build-k8s-all
+    rules:
```
Comment (a-wai) [inline config/pipeline.yaml:1679]: Actually I assume we also want to apply this rule to `kbuild-gcc-12-riscv` as well, right? This job [got triggered](https://staging.kernelci.org:9000/viewer?node_id=66b20bee2fbd8114fc0b7438) on `efi` (worked fine to exclude ARC/MIPS builds though, and `kbuild-gcc-12-riscv-build-only` as well)

**Review by a-wai — CHANGES_REQUESTED** [review]

File: config/pipeline.yaml (hunk 3 above)
Comment (musamaanjum) [reply config/pipeline.yaml:1682]: Done

**Review by a-wai — APPROVED** [review]

**Pull request merged**